### ISSUE TITLE
Start of Week not applied to Field Filter in Native question, which can lead to incorrect results

### ISSUE THREAD

**Issue by ellie-metabase** [metabase/metabase#14294]
**Describe the bug**
Regardless of which day "start of week" is configured to, using the Field Filter in a Native Query will return results with Sunday as the start of the week.

**To Reproduce**

This bug is in current build, v1.37.3
1. Go to Settings > Admin > Localization
2. Change "First Day of Week" to Monday
3. Click "Ask A Question" > "Native Query" 
4. Select the Sample Dataset
5. Enter the following query
```
select ID, CREATED_AT, dayname(CREATED_AT) as CREATED_AT_DAY
from ORDERS 
[[where {{date_range}}]]
order by CREATED_AT
```
6. Configure the Variables Settings as follows:
Variable type: Field Filter
Field to map to: Order > Created At
Filter widget type: Date Filter
7. Set the filter to include the previous 52 weeks

**Expected behavior**
Since this sample dataset has a record for every day between Jan 1, 2020 - April 19, 2020 the oldest record for this filter should match the day of week we specified in "Start of Week".  

**Observed Behavior**
Regardless of what I configure the "Start of Week" day to, the filter always considers Sunday the start of the week.

**Screenshots**
<img width="1827" alt="startsatsunday" src="https://user-images.githubusercontent.com/74572342/103955085-c7467500-5102-11eb-9827-f9f0a6a2804e.png">

At the time of running this query, the start of "Previous 52 Weeks" should be Monday, Jan 6 2020.  However, the query results include Sunday, Jan 5th.  

**Information about your Metabase Installation:**
v1.37.3, docker running locally

**Severity**
This affects weekly reporting for customers that have their start of week set to something other than Sunday.  

Redshift treats Monday as the first day of the week in DATE_TRUNC, so this incongruity causes visualizations to have an extra week with values incorrectly offset in groupings.

**Comment by nemanjaglumac** [comment]
The key part of the repro steps is:
> **At the time of running this query**, the start of "Previous 52 Weeks" should be Monday, Jan 6 2020. However, the query results include Sunday, Jan 5th.

As the time goes on, the test will change to the point where the previous 52 weeks don't yield any result. @flamber is there any other way to reproduce this more reliably (other than brute-forcing it with something like "previous 1000 weeks")?

**Comment by flamber** [comment]
@nemanjaglumac No, and don't think you can even reproduce with "previous 1000 weeks". It would need a custom schema, so you have control over the date column (like always setting it to last week) and then you could use "previous 1 week".

**Comment by nemanjaglumac** [comment]
Previous 220 weeks currently gives us October 16, 2016 (Sunday) instead of October 17, 2016 (Monday)
![image](https://user-images.githubusercontent.com/31325167/104040999-a3556300-51d8-11eb-87ae-d1f8ef8f4829.png)

If you try decreasing that number of weeks to simulate the passing of time, you'll notice it always shows Sunday within the date range in sample dataset. This gives us 3+ years of "test validity". Are you ok with me reproducing this and making a TODO comment to disable or update this test in 2 years or so?

IMO, the benefit of having a reproduction when devs try to fix this issue probably outweighs the negatives of its fragility.

**Comment by flamber** [comment]
@nemanjaglumac But next week, you will not get those results. This is the core problem (see the row count)
![image](https://user-images.githubusercontent.com/1447303/104042353-b10be800-51da-11eb-93fc-424841c5ed0b.png)

**Comment by nemanjaglumac** [comment]
But I can always assert that the first row starts on Sunday/Monday, regardless of the row count. See the screenshot - no matter how many weeks in the past I go, the first row will always be Sunday. 
![image](https://user-images.githubusercontent.com/31325167/104049836-e4a03f80-51e5-11eb-8936-6b7810a50ee8.png)

Asserting on Sunday rather than any strict date gives me the flexibility I need (unless I am missing something else?).

**Comment by flamber** [comment]
As long as there's an order on every Monday for all the weeks, then it will probably be fine. You should probably add a "TODO", so the test gets a revamp, once issue 4083 is implemented.

**Comment by ellie-metabase** [comment]
I didn't check the entire data set but for what it's worth, there is
definitely an order for every single day for the period of time Jan 1, 2020
- Jan 1, 2020 - April 19, 2020

**Comment by camsaul** [comment]
Fixed by #15040
